### PR TITLE
TTS v1.9

### PR DESCRIPTION
**Pull request by Bod9001**
TODOs
so, still not set up with the proper audio channels and player tracking as the audio source live on the player I mean
and
Would like to add an option to set the default voice if one isn't specified in the settings ( for stuff like automated messages that haven't got it specified   )  /
and  code scan White list /
add test to stop you putting invalid voices  /

### Changelog:

CL: [New] new and improved TTS voices
CL: [New] you can now select what voice your character has
CL: [Improvement] The game now uses a Locally Hosted TTS engine for TTS

### CODE REVIEW

File: UnityProject/Assets/Scripts/Core/SecureStuff/TTSCommunication.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+		private class DataTSS
+		{
+			public string input_string { get; set; }
+			public string voice { get; set; }
+		}
+
+		public static async Task<byte[]> GenTTS(string Input, string voice)
+		{
+			// URL of the API endpoint
+			string url = "http://127.0.0.1:5234/generate-audio";
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/Core/SecureStuff/TTSCommunication.cs:22]: shouldn't this be pre-configured in the game config?

File: UnityProject/Assets/Scripts/Core/Utils/MultiInterestString.cs [hunk 2]
```diff
@@ -0,0 +1,6 @@
+using UnityEngine;
+
+public class MultiInterestString
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/Core/Utils/MultiInterestString.cs]: leftover?

**Review by MaxIsJoe — CHANGES_REQUESTED** [review]

File: UnityProject/Assets/Scripts/Core/SecureStuff/TTSCommunication.cs (hunk 1 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Core/SecureStuff/TTSCommunication.cs:22]: will always be using local server and this is secure stuff so shouldn't be able to be modified

File: UnityProject/Assets/Scripts/Core/Utils/MultiInterestString.cs (hunk 2 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Core/Utils/MultiInterestString.cs]: well is for when tongues have voices, in case you have multiple tongues xD

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Core/Utils/MultiInterestString.cs]: so its not being used at the moment?

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Core/Utils/MultiInterestString.cs]: yeah

File: UnityProject/Assets/Scripts/Core/TTS/TTSVoices.cs [hunk 3]
```diff
@@ -0,0 +1,74 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+public static class TTSVoices
+{
+	public static List<string> Voices = new List<string>()
+	{
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/Core/TTS/TTSVoices.cs:8]: this needs to be dynamically generated at some point, but tis good for now

**Review by MaxIsJoe — DISMISSED** [review]
LGTM

**Pull request merged**